node/module.ts

```diff
@@ -210,10 +210,15 @@ class Module {
   static wrapper = [
     // We provide non standard timer APIs in the CommonJS wrapper
     // to avoid exposing them in global namespace.
-    "(function (exports, require, module, __filename, __dirname, setTimeout, clearTimeout, setInterval, clearInterval) { ",
-    "\n});",
+    `
+    (function (exports, require, module, __filename, __dirname, setTimeout, clearTimeout, setInterval, clearInterval) {
+      (function (exports, require, module, __filename, __dirname) {
+    `,
+    `
+      }).call(this, exports, require, module, __filename, __dirname);
+    })
```

Comment:
I'm not a fan of this solution - this will completely break the source map support in CJS files; I think reverting injection of timers globals is a better solution.

Just FYI in Node's CJS loader, there's a mechanism that checks if wrapper was "patched" and then the loader behaves slightly differently.

Reply:
> this will completely break the source map support in CJS files

Can you elaborate how this breaks the cjs source maps? (Do we support that? If we make this wrapper a single line, can it be a workaround?)

> I think reverting injection of timers globals is a better solution.

I think reverting is not an option at the moment as it breaks the yarn compatibility. Yarn uses global `setInterval().unref()` [here](https://github.com/yarnpkg/yarn/blob/3119382885ea373d3c13d6a846de743eca8c914b/src/reporters/base-reporter.js#L169)

Reply:
> Can you elaborate how this breaks the cjs source maps? (Do we support that? If we make this wrapper a single line, can it be a workaround?)

Deno in not aware of this wrapper so when applying source maps it doesn't know that it should perform some additional calculations to take it into account. This more or less works fine if the wrapper is on a single line (you might get an information in the stack trace that position was out of bounds). But if we make it multiline, then it's pretty much guaranteed that stack traces will print wrong lines

Reply:
Thanks for the explanation! Then I made it a one-liner. What do you think about the current version?

Reply:
It's better than multiline :) let's land it

Reply:
Thanks!